pkg/trace/agent/agent.go

```diff
@@ -446,16 +446,19 @@ func (a *Agent) runSamplers(now time.Time, pt traceutil.ProcessedTrace, hasPrior
 // ErrorSampler are run in parallel. The RareSampler catches traces with rare top-level
 // or measured spans that are not caught by PrioritySampler and ErrorSampler.
 func (a *Agent) samplePriorityTrace(now time.Time, pt traceutil.ProcessedTrace) bool {
+	var rare bool
+	if a.conf.DisableRareSampler {
+		rare = false
+	} else {
+		rare = a.RareSampler.Sample(now, pt.TraceChunk, pt.TracerEnv)
+	}
```

Comment:
This could actually be a one liner, but I thought it wasn't super readable. Let me know:
```
rare := !a.conf.DisableRareSampler && a.RareSampler.Sample(now, pt.TraceChunk, pt.TracerEnv)
```

Reply:
I agree your version is better.